lib/tcp/tcp_packet.ml

```diff
@@ -73,7 +73,6 @@ module Marshal = struct
   let unsafe_fill ~pseudoheader ~payload t buf options_len =
     let data_off = (sizeof_tcp / 4) + (options_len / 4) in
     let buf = Cstruct.sub buf 0 (data_off * 4) in
-    Cstruct.memset buf 0; (* autopad *)
```

Comment:
looks like this got added in 4f3acd9968e12843d5134b4d28b around the time that Cstruct zeroing policy also got stricter, hence the confusion. We should add a test case for checking this doesnt regress as well - I can do that tomorrow if you dont have one already @balrajsingh